app/features/receive/receive-cashu-token.tsx

```diff
@@ -22,29 +29,65 @@ type Props = {
 export default function ReceiveToken({ token }: Props) {
   const [_, copyToClipboard] = useCopyToClipboard();
   const { toast } = useToast();
-  const {
-    data: tokenData,
-    isClaiming,
-    handleClaim,
-    setReceiveAccount,
-  } = useReceiveCashuToken({
+  const addCashuAccount = useAddCashuAccount();
+  const { claimableToken, cannotClaimReason } = useTokenWithClaimableProofs({
     token,
-    cashuPubKey:
+    cashuPubKey: // TODO: replace with user's pubkey from OS
       '038127ae202c95f4cd4ea8ba34e73618f578adf516db553a902a8589796bdc373',
   });
-
+  const sourceAccount = useCashuTokenSourceAccount(token);
   const {
-    receiveAccount,
     selectableAccounts,
-    crossMintSwapDisabled,
+    receiveAccount,
     receiveAccountIsSource,
-    cannotClaimReason,
-    isMintKnown,
-    claimableToken,
-  } = tokenData;
+    isSourceAccountAdded,
+    isCrossMintSwapDisabled,
+    setReceiveAccount,
+  } = useReceiveCashuTokenAccounts(sourceAccount);
+
+  const { status, claimToken } = useReceiveCashuToken({
+    onError: (error) => {
+      toast({
+        title: 'Failed to claim token',
+        description: error.message,
+        variant: 'destructive',
+      });
+    },
+  });
+
+  const handleClaim = async () => {
+    if (!claimableToken) {
+      return;
+    }
+
+    let account: CashuAccount = receiveAccount;
+
+    if (receiveAccount.id === '') {
+      try {
+        account = await addCashuAccount(receiveAccount);
+      } catch (error) {
+        console.error(error);
+        return toast({
+          title: 'Failed to add account',
+          description: 'Please try again',
+        });
+      }
+    }
+
+    await claimToken({ token, account });
```

Comment:
what happens when this fails if account was added but claim token fails and then you try again? will it try to add the account that was already added again?

Reply:
I think it would be better to only have `setReceiveAccount` method which checks if the account is added and if not adds it and that method can use for all cases.